File: romain-to-integer/romain-to-integer.go
func romanToInt(s string) int {
    dict := map[byte]int{
		  'I': 1,
		  'V': 5,
		  'X': 10,
		  'L': 50,
		  'C': 100,
		  'D': 500,
		  'M': 1000,
	  }

    result := 0
    for i := 0; i < len(s); i ++ {
        if (i + 1 != len(s) && dict[s[i]] < dict[s[i+1]]) {
            result -= dict[s[i]]
        } else {
            result += dict[s[i]]
        }
    }

    return result
}

File: leetcode/a-easy/romain-to-integer/romain-to-integer.ts
function romanToInt(str: string): number {
    const rep: {[key: string]: number} = {
        I: 1,
        V: 5,
        X: 10,
        L: 50,
        C: 100,
        D: 500,
        M : 1000,
    }

    let sum = 0;
    for (let i = 0; i < str.length; i++) {
        if (i + 1 < str.length && rep[str[i]] < rep[str[i + 1]]) {
            sum -= rep[str[i]];
        } else {
            sum += rep[str[i]];
        }
    }

    return sum;
};
